sass/sass.bzl

```diff
@@ -11,151 +11,114 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"Compile Sass files to CSS"
 
-SASS_FILETYPES = FileType([
-    ".sass",
-    ".scss",
-])
+_FILETYPES = [".sass", ".scss", ".svg", ".png", ".gif"]
 
-def collect_transitive_sources(ctx):
-    source_files = depset(order="postorder")
-    for dep in ctx.attr.deps:
-        source_files += dep.transitive_sass_files
-    return source_files
+SassInfo = provider("transitive_sources")
+
+def collect_transitive_sources(srcs, deps):
+    return depset(
+        srcs,
+        transitive=[dep[SassInfo].transitive_sources for dep in deps],
+        order="postorder")
 
 def _sass_library_impl(ctx):
-    transitive_sources = collect_transitive_sources(ctx)
-    transitive_sources += SASS_FILETYPES.filter(ctx.files.srcs)
-    return struct(
-        files = depset(),
-        transitive_sass_files = transitive_sources)
+    transitive_sources = collect_transitive_sources(
+        ctx.files.srcs, ctx.attr.deps)
+    return [SassInfo(transitive_sources=transitive_sources)]
 
 def _sass_binary_impl(ctx):
-    # Reference the sass compiler and define the default options
-    # that sass_binary uses.
-    sassc = ctx.file._sassc
-    options = [
-        "--style={0}".format(ctx.attr.output_style),
-        "--sourcemap",
-    ]
-
-    # Load up all the transitive sources as dependent includes.
-    transitive_sources = collect_transitive_sources(ctx)
-    for src in transitive_sources:
-        options += ["--load-path", src.path[:-len(src.basename)]]
-
-    ctx.action(
-        inputs = [sassc, ctx.file.src] + list(transitive_sources),
-        executable = sassc,
-        arguments = options + [ctx.file.src.path, ctx.outputs.css_file.path],
+    args = ctx.actions.args()
+    args.add(["--style", ctx.attr.output_style], join_with="=")
+    for prefix in [".", ctx.var['BINDIR'], ctx.var['GENDIR']]:
```

Comment:
Add a comment like?
```sh
# Sources for compilation may exist in the source tree, in bazel-bin, or bazel-genfiles. 
# Additionally, the Sass compiler allows extra "include_paths" where it will look for sources.
# We include those include_paths for source, bazel-bin, and bazel-genfiles as well.
```